src/dimension_slice.c

```diff
@@ -147,7 +151,7 @@ dimension_slice_scan_limit_internal(int indexid, ScanKeyData *scankey, int nkeys
  * Returns a dimension vector of slices that enclose the coordinate.
  */
 DimensionVec *
-ts_dimension_slice_scan_limit(int32 dimension_id, int64 coordinate, int limit)
+ts_dimension_slice_scan_limit(int32 dimension_id, int64 coordinate, int limit, ScanTupLock *tuplock)
```

Comment:
Note that the "tuple found" function below (`dimension_vec_tuple_found`) should check the lock result (when a tuple lock is requested) in case the tuple lock is not successful. 

This applies to all tuple handler functions where tuple locks are a possibility.

Reply:
Tuple lock checks added to the two functions that we use.